src/event-handlers/message.handler.ts

```diff
@@ -14,23 +15,24 @@ export class MessageHandler implements IEventHandler<MessageHandler['EVENT_NAME'
         if (message.author.bot) return;
         if (message.content.indexOf(config.botConfig.prefix)) return;
 
-        const args = message.content.slice(config.botConfig.prefix.length).split(/ +/g);
-        const command: string = args.shift()?.toLowerCase() || 'default';
+        const args: string[] = message.content.slice(config.botConfig.prefix.length).split(/ +/g);
+        const keyword: string = args.shift()?.toLowerCase() || 'default';
 
         // TODO: add check for moderation commands -- should bypass rate limit
         if (this.commandRateLimiter.shouldReject(message.author))
             return void message.channel.send('You are sending too many messages!');
 
-        const toExecute: ICommand = commands.find((cmd: ICommand) => {
-            cmd.triggers.some((t: Trigger) => {
-                if (typeof t === 'function')
-                    return t(message, command, args);
-                if (t instanceof RegExp)
-                    return t.test(message.content);
-                return command === t;
-            });
-        }) || defaultCommand;
-
-        toExecute.execute(message, command, args);
+
+        const [command, trigger] = this.getTriggeredCommand(message, keyword, args) ?? [defaultCommand, NullTrigger];
```

Comment:
Smart - basically taking the triggeredCommand and returning defaultCommand if get...() returns null!